### DISCUSSION TITLE
Block_Break particle in Minestom

### DISCUSSION THREAD

**Discussion by KianBrose** [Minestom/Minestom#1796] [Help] [answered]
Context: I am currently implementing guns into minestom, the bullet itself is a EntityProjectile which triggers the `ProjectileCollideWithBlockEvent`. When it collides with a block, I want to display particles as if that block was breaking, as seen in these images from a PaperMC implementation:
![image](https://user-images.githubusercontent.com/62078964/226346577-425adc01-870f-42fb-bc41-6783e01bf7f5.png)
![image](https://user-images.githubusercontent.com/62078964/226346593-f91846db-5d8d-4be6-907d-e65cb7d802c4.png)
![image](https://user-images.githubusercontent.com/62078964/226346614-757a7eef-583c-4276-89a2-85fd33dc12d4.png)
I have tried figuring out how to do this on minestom for the past 3 days with no luck, any clues?

**Comment by p3sto** [comment]
Try getting the `Block` from the event and the spawn the particle type `BLOCK` passing that as the parameter

**Reply by reeve567** [reply]
Just now realizing this was here when I replied, whoops

**Comment by reeve567** [comment]
Mentioned this in discord, but might as well mention it here: I think you're looking for the `minecraft:block` particle
![image](https://user-images.githubusercontent.com/18086052/226407031-385db52a-19b5-4bb2-9c80-55dc89219464.png)
Or you could try using `minecraft:falling_dust`

**Comment by KianBrose** [comment, marked as answer]
Solution was:
```
event.getEntity().sendPacketToViewers(
        ParticleCreator.createParticlePacket(
                Particle.BLOCK,
                false,
                event.getEntity().getPosition().x(),
                event.getEntity().getPosition().y(),
                event.getEntity().getPosition().z(),
                0.2f,
                0.2f,
                0.2f,
                0f,
                30,
                binaryWriter -> {
                    binaryWriter.writeVarInt(Block.REDSTONE_BLOCK.stateId());
                }
        )
);
```
